shared/actions/chat/index.js

```diff
@@ -1016,6 +1016,33 @@ function* _exitSearch() {
   yield put(Creators.setInboxFilter([]))
 }
 
+function* _selectSearchResultWithMovement({payload: {movement}}: Constants.SelectSearchResultIdWithMovement) {
+  const searchResults = yield select(({chat: {searchResults}}: TypedState) => searchResults)
```

Comment:
[Discussion, can defer from review]: I think that this sort of operation should be done in the reducer, if possible. It's possible for something to run in between these two selectors, causing them to be out of sync. Might be better to combine them into a single selector, but still feels like doing them outside of the saga asyncness is safer.